src/redux/Rockets/rockets.js

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const LODING = 'LODING';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+export const loadRockets = createAsyncThunk(LODING, async () => {
+  const response = await axios.get(url);
+  const res = response.data;
+  const data = res.map((item) => ({
+    id: item.id,
+    name: item.rocket_name,
+    description: item.description,
+    img: item.flickr_images[0],
+    value: false,
+  }));
+  return data;
+});
+
+const storeSlice = createSlice({
+  name: 'space Hup',
+  initialState: [],
+  reducers: {
+    rocketReserve(state, action) {
+      const newState = state.map((rocket) => {
+        if (rocket.id === Number(action.payload.id)) {
+          return {
+            ...rocket,
+            value: !rocket.value,
+          };
+        }
+        return {
+          ...rocket,
+        };
```

Comment:
- [ ] In this line, is it possible to return only `rocket` instead of `{...rocket}`? If so please change this line :+1:

Reply:
it will be done in the final repo @karlavdelgadof

Reply:
Ok @Alzubair98 got it! thanks for the explanation :+1: